### PR TITLE
OCPBUGS-17724: Graceful fail for AWS getUser on destroy

### PR DESCRIPTION
**Pull request by barbacbd**
**The getUser role validation can be skipped during manual mode installations. During the uninstall process, fail gracefully when the role is not found.

### CODE REVIEW

File: pkg/destroy/aws/iamhelpers.go [hunk 1]
```diff
@@ -112,13 +112,24 @@ func (search *iamUserSearch) arns(ctx context.Context) ([]string, error) {
...
+					var awsErr awserr.Error
+					if errors.As(err, &awsErr) {
+						switch {
+						case awsErr.Code() == iam.ErrCodeNoSuchEntityException:
+							// The role does not exist.
+							// Ignore this IAM Role and do not report this error via lastError.
+							search.unmatched[*user.Arn] = exists
+						case strings.Contains(err.Error(), "AccessDenied"):
+							// Installer does not have access to this IAM role.
+							// Ignore this IAM Role and do not report this error via lastError.
```
Comment (rna-afk) [inline pkg/destroy/aws/iamhelpers.go:125]: How about a log line here? What if they did provide a credentials in a normal install that was rejected.

Comment (sadasu) [reply pkg/destroy/aws/iamhelpers.go:125]: That log line will appear multiple times in the logs. This fix is based on https://github.com/openshift/installer/pull/7180 and the "Actual results" section of the associated issue https://issues.redhat.com//browse/OCPBUGS-1769 has sample output.

Comment (rna-afk) [reply pkg/destroy/aws/iamhelpers.go:125]: Some user will in the future definitely want to know why the creds they provided not work. I'm fine if you all are worried about log lines repeating.

Comment (sadasu) [reply pkg/destroy/aws/iamhelpers.go:125]: We can update code to make sure the message is logged only once. It just can't be within the switch block.

**Pull request merged**